libs/wingc/src/jsify/snapshots/access_methods_and_properties_on_collections.snap

```diff
@@ -48,6 +48,8 @@ const $wing_is_test = process.env.WING_IS_TEST === "true";
 const std = $stdlib.std;
 const $helpers = $stdlib.helpers;
 const $extern = $helpers.createExternRequire(__dirname);
+const $PlatformManager = new $stdlib.platform.PlatformManager({platformPaths: $platforms});
+globalThis.$PolyconFactory = $PlatformManager.createPolyconFactory();
```

Comment:
I still think we have too many layers and concepts here. Let's take this opportunity to remove redundant abstractions...

I would combine `PlatformManager` and `PolyconFactory` into a single concept.

Then:

```suggestion
globalThis.$platform = $stdlib.platform.load({ platformPaths: $platforms });
```

Then:

```js
globalThis.$platform.resolveType(fqn)
```


This will simplify all the hula hoops you need to do with .bind() and factories and managers and all that jazz.

What do you think?
E

Reply:
I’m unsure about collapsing the factory and platform manager. Maybe it’s the right call atm, but there’s still missing pieces of the platform system to implement, like platform compatibility which is an issue we also saw when different platforms have expectations on the type of App or provisioning engine

Also there is the discussion we have been having around defining differences in platforms and platform extensions. Maybe a manager isn’t that helpful with this (haven’t put much thought) but it might be better to leave for now

Reply:
Some of these directions are approached in https://github.com/winglang/wing/pull/6759 -- for now I'm going to try and scope the changes to just those that are needed for #6490